optimize/backend/src/main/java/io/camunda/optimize/rest/security/cloud/CCSaaSSecurityConfigurerAdapter.java

```diff
@@ -105,6 +112,21 @@ public AuthenticationCookieFilter authenticationCookieFilter(final HttpSecurity
             .build());
   }
 
+  @Bean
+  @Order(Ordered.HIGHEST_PRECEDENCE) /* order of loading */
+  FilterRegistrationBean<SaasRequestAdjustmentFilter> requestAdjuster() {
+    LOG.debug("Registering filter 'requestAdjuster' (SaaS)...");
+    final SaasRequestAdjustmentFilter saasRequestAdjustmentFilter =
+        new SaasRequestAdjustmentFilter(
+            configurationService.getAuthConfiguration().getCloudAuthConfiguration().getClusterId());
+    final FilterRegistrationBean<SaasRequestAdjustmentFilter> registration =
+        new FilterRegistrationBean<>();
+    registration.setOrder(Ordered.HIGHEST_PRECEDENCE); /* position in the chain */
+    registration.addUrlPatterns("/*");
+    registration.setFilter(saasRequestAdjustmentFilter);
+    return registration;
+  }
+
   @Bean
   @Order(1)
   protected SecurityFilterChain configurePublicApi(final HttpSecurity http) {
```

Comment:
out of interest, did you test the public API?

Reply:
I did some local tests in self-managed as well as some tests against the cluster.
I listed the API  as one of the things to QA in the description

Reply:
and to confirm, you mean the public API and not the shared report/dashboard functionality?

Reply:
To my knowledge, the shared functionality uses the API with the `/external/api/` prefix.
I tried to execute:
* /external/api/ui-configuration
* /api/external/ui-configuration
* /api/ui-configuration (this should be public)

and everything worked, in self-managed and against the cluster. I did some other tests with the readyz endpoint, but not more extended than this. So I pretty much made sure that the requests are forwarded to the correct handler, but a more exhaustive test set should be done, to make sure all of the APIs work.

If we're good with time, an integration test could be good

Reply:
These are non-authenticated parts of the API. There are also parts of our API [here](https://docs.camunda.io/optimize/next/apis-tools/optimize-api/overview/), like external variable ingestion.

I am ok with this being pushed to QA to validate, I just want to ensure that it is not missed